From d182738c325fec06cfe43a4c6b0567b58402dc9a Mon Sep 17 00:00:00 2001
From: holishing <holishingtest@gmail.com>
Date: Mon, 31 Jul 2017 20:34:38 +0800
Subject: [PATCH 03/10] Solving a PERM bug violately. need to be checked

---
 src/maple/mail.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/maple/mail.c b/src/maple/mail.c
index e2c262b..b535547 100755
--- a/src/maple/mail.c
+++ b/src/maple/mail.c
@@ -1141,7 +1141,8 @@ m_zip()			/* itoc.010228: 
       return XEASY;
     }
 
-    if ((ans == '3' && !(bbstate & STAT_BOARD)) || (ans == '4' && !(bbstate & STAT_BOARD)))
+    if ((ans == '3' && !(bbstate & STAT_BOARD)) || (ans == '4' && !(bbstate & STAT_BOARD))) 
+                             /*r2: STAT_BOARD ==> STAT_BM , Solving a PERM bug violately, need to be checked*/
     {
       vmsg("只有板主才能打包看板文章及看板精華區");
       return XEASY;
-- 
1.8.3.1

