From 9db417aa5bbc3ef44e2ea21a297d439ff045cb64 Mon Sep 17 00:00:00 2001
From: holishing <holishingtest@gmail.com>
Date: Tue, 8 Aug 2017 18:05:05 +0800
Subject: [PATCH 10/10] for solving SegFault

---
 src/maple/mail.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/maple/mail.c b/src/maple/mail.c
index b4d414d..ca95a33 100755
--- a/src/maple/mail.c
+++ b/src/maple/mail.c
@@ -1056,9 +1056,8 @@ do_forward(title, mode)
   int rc;
   char *userid;
   char addr[64], fpath[64], cmd[256];
-/*
   strcpy(addr, cuser.email);
-
+/*
   if (!vget(b_lines, 0, "請輸入轉寄地址：", addr, 60, GCARRY))
     return;
 
-- 
1.8.3.1

