From 1da579d9ddb2a6a29d2fb7675f3b419526f979d5 Mon Sep 17 00:00:00 2001
From: holishing <holishingtest@gmail.com>
Date: Tue, 8 Aug 2017 17:12:42 +0800
Subject: [PATCH 09/10] cuser.email only

---
 src/maple/mail.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/maple/mail.c b/src/maple/mail.c
index b535547..b4d414d 100755
--- a/src/maple/mail.c
+++ b/src/maple/mail.c
@@ -1056,7 +1056,7 @@ do_forward(title, mode)
   int rc;
   char *userid;
   char addr[64], fpath[64], cmd[256];
-
+/*
   strcpy(addr, cuser.email);
 
   if (!vget(b_lines, 0, "請輸入轉寄地址：", addr, 60, GCARRY))
@@ -1069,6 +1069,8 @@ do_forward(title, mode)
   }
 
   sprintf(fpath, "確定寄給 [%s] 嗎(Y/N)？[N] ", addr);
+*/
+  sprintf(fpath, "確定寄給 [%s] 嗎(Y/N)？[N] ",cuser.email);
   if (vans(fpath) != 'y')
     return;
 
@@ -1123,7 +1125,7 @@ m_zip()			/* itoc.010228: 
   int ans;
   char *name, *item, buf[80];
 
-  ans = vans("打包資料 1)個人信件 2)個人精華區 3)看板文章 4)看板精華區 [Q] ");
+  ans = vans("打包資料 1)個人信件 2)個人精華區 3)看板文章 4)看板精華區 到註冊信箱 [Q] ");
 
   if (ans == '1' || ans == '2')
   {
@@ -1141,8 +1143,8 @@ m_zip()			/* itoc.010228: 
       return XEASY;
     }
 
-    if ((ans == '3' && !(bbstate & STAT_BOARD)) || (ans == '4' && !(bbstate & STAT_BOARD))) 
-                             /*r2: STAT_BOARD ==> STAT_BM , Solving a PERM bug violately, need to be checked*/
+    if ((ans == '3' && !(bbstate & STAT_BM)) || (ans == '4' && !(bbstate & STAT_BOARD))) 
+                             /*r2: STAT_BOARD or STAT_BM ? need to be checked*/
     {
       vmsg("只有板主才能打包看板文章及看板精華區");
       return XEASY;
-- 
1.8.3.1

